### PR TITLE
Ag 7746/log axis domain

### PR DESCRIPTION
**Pull request by manapeirov**
PR for https://ag-grid.atlassian.net/browse/AG-7746

### CODE REVIEW

File: charts-packages/ag-charts-community/src/axis.ts [hunk 1]
```diff
@@ -1032,7 +1033,9 @@ export class Axis<S extends Scale<D, number>, D = any> {
 
     // For formatting (nice rounded) tick values.
     formatTickDatum(datum: any, index: number): string {
-        const { label, labelFormatter, fractionDigits } = this;
+        const { label, labelFormatter, fractionDigits, scale } = this;
+
+        const logScale = scale instanceof LogScale;
 
         return label.formatter
```
Comment (alantreadway) [inline charts-packages/ag-charts-community/src/axis.ts:1040]: The logic here is quite unreadable, would you mind splitting these nested ternary statements this into a simple `if ... else if ... else ...` block?

File: charts-packages/ag-charts-community/src/util/array.ts [hunk 2]
```diff
@@ -33,17 +33,25 @@ export function extent(values: Array<number | Date>): [number, number] | undefin
  * finds the min and max using a process appropriate for stacked values. Ie,
  * summing up the positive and negative numbers, and returning the totals of each
  */
-export function findMinMax(values: number[]): { min?: number; max?: number } {
+export function findMinMax(values: number[]): { min?: number; max?: number; extent?: [number, number] } {
```
Comment (alantreadway) [inline charts-packages/ag-charts-community/src/util/array.ts:36]: We should really rename this to `findMinMaxExtent`?

Comment (alantreadway) [reply charts-packages/ag-charts-community/src/util/array.ts:36]: Should we be combining the responsibilities for `min`, `max` and `extent` though? Seems like we end up with two distinct (and later divergent `extent` calculations?

**Comment by manapeirov** [comment]
@alantreadway PTAL

**Pull request merged**